security

calebrwalk5 opened the pull request.

calebrwalk5 committed:
add handling for mysqldump failure

calebrwalk5 committed:
verify client ip

calebrwalk5 committed:
verify client ip

Itz-Hyperz commented on website/backend.js:
```diff
@@ -26,7 +25,11 @@ async function init(app) {
-    app.use(requestIp.mw());
+    app.use((req, res, next) => {
+        const clientIp = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
+        console.log(`Client IP: ${clientIp}`);
```
if(config.debugMode) console.log(`Client IP: ${clientIp}`);

calebrwalk5 replied:
smart man

Itz-Hyperz committed:
yeah lol

https://cdn.hyperz.net/u/main/TwVbi59.png
https://cdn.hyperz.net/u/main/5djXT8G.png

Itz-Hyperz committed:
Update index.js

Itz-Hyperz committed:
callbacks wouldn't work plus you cant return headers to the client more than once.

calebrwalk5 commented:
like i said @Itz-Hyperz  i am an idea machine, i give good ideas not code

Itz-Hyperz committed:
moved so it's defined properly

https://cdn.hyperz.net/u/main/PTxinFU.png

Itz-Hyperz commented:
not entirely sure mysqldump.on('error') would ever be emitted, but just in case, and to make you feel better, out of pity, I will merge :]
